### PR TITLE
feature: add device option for container

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: HusterWan <zirenwan@gmail.com>

**1.Describe what this PR did**
Support mapping host's devices to container when creates a container

**2.Does this pull request fix one issue?** 

**3.Describe how you did it**

**4.Describe how to verify it**
```
[root@docker-registry github.com]# pouch run --device /dev/null:/dev/testnull:rwm  registry.hub.docker.com/library/busybox:latest ls /dev/testnull
/dev/testnull
```

**5.Special notes for reviews**

### CODE REVIEW

**Comment by allencloud** [comment]
Thanks for your work, @HusterWan 
While I have one confusion for us, here we have vendored package 

github.com/opencontainers/runc/libcontainer/configs
github.com/opencontainers/runc/libcontainer/devices
github.com/opencontainers/runc/libcontainer/user
github.com/opencontainers/runc/libcontainer/system

But the previous two packages have difference commit ID, and they are from same package libcontainer, and from difference deep package. Do we need to unify the commit ID of these packages? @HusterWan @skoo87 @rudyfly

**Comment by allencloud** [comment]
/cc @WIZARD-CXY Please check this if the device options implemented is what you want.
Check and test if the functionality is sufficient for you.  Thanks.

**Comment by skoo87** [comment]
@HusterWan  please update #463

File: cli/create.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ func (cc *CreateCommand) addFlags() {
 	flagSet.Int64Var(&cc.cpushare, "cpu-share", 0, "CPU shares")
 	flagSet.StringVar(&cc.cpusetcpus, "cpuset-cpus", "", "CPUs in cpuset")
 	flagSet.StringVar(&cc.cpusetmems, "cpuset-mems", "", "MEMs in cpuset")
+	flagSet.StringSliceVarP(&cc.devices, "--device", "", nil, "Add a host device to the container")
```
Comment (allencloud) [inline cli/create.go]: s/--device/device 

I am afraid `--device` is not correct 😕

**Comment by HusterWan** [comment]
Updated, @allencloud @Letty5411 BTW, i can not edit the page #463

**Comment by HusterWan** [comment]
@allencloud  Update github.com/opencontainers/runc/libcontainer commit number to e6516b3d5dc780cb57a976013c242a9a93052543
if there is any problem, please let me know at any time, thanks.

**Comment by Letty5411** [comment]
LGTM

File: cli/container.go [hunk 2]
```diff
@@ -90,6 +91,71 @@ func (c *container) config() (*types.ContainerCreateConfig, error) {
...
+		}
+		deviceMappings = append(deviceMappings, deviceMapping)
+	}
+	config.HostConfig.Devices = deviceMappings
 
 	return config, nil
 }
+
+// parseDevice parses a device mapping string to a container.DeviceMapping struct
+func parseDevice(device string) (*types.DeviceMapping, error) {
```
Comment (allencloud) [inline cli/container.go]: I think it is better to add some unit tests for function `parseDevice` and `validDeviceMode`.

In addition, this validation is good in client side, while maybe in the future we should support this in daemon side. We need to validate the API.

Comment (HusterWan) [reply cli/container.go]: Yes, your suggestion is great, i will put this two function into pkg/utils, and add some unit tests

**Comment by skoo87** [comment]
Please update the list https://github.com/alibaba/pouch/issues/463 @HusterWan

**Comment by allencloud** [comment]
LGTM except unit tests are added.

**Comment by HusterWan** [comment]
@allencloud Sorry, yesterday was too busy to add the unit test, i will finish today

**Comment by HusterWan** [comment]
Updated, PTAL thanks, @allencloud @Letty5411

**Comment by allencloud** [comment]
LGTM， thanks a lot for your work. @HusterWan 
And please check this and tell us if this could meet your GPU device's requirement. @WIZARD-CXY Thanks.

**Pull request merged**